docs/reference/mapping/types/json.asciidoc

```diff
@@ -0,0 +1,200 @@
+[[json]]
+=== JSON datatype
+
+experimental[The `json` field type is experimental and may be changed in a breaking way in future releases.]
+
+By default, each subfield in an object is mapped and indexed separately. If
+the names or types of the subfields are not known in advance, then they are
+<<dynamic-mapping, mapped dynamically>>.
+
+The `json` type provides an alternative approach, where the entire object is
+mapped as a single field. Given an object, the `json` mapping will parse out
+its leaf values and index them into one field. The object's contents can then
+be searched through simple keyword-style queries.
+
+This data type can be useful for indexing objects with a very large number of
+distinct keys. Compared to mapping each field separately, `json` fields have
+the following advantages:
+
+- Only one field mapping is created for the whole object, which can help
+  prevent a <<mapping-limit-settings, mappings explosion>> due to a large
+  number of field mappings.
+- A `json` field may take up less space in the index, as only one underlying
+  field is created.
+
+However, `json` fields present a trade-off in terms of search functionality.
+Only basic queries are allowed, with no support for numeric range queries or
+aggregations. Further information on the limitations can be found in the
+<<supported-operations, Supported operations>> section.
+
+NOTE: The `json` mapping type should **not** be used for indexing all JSON
+content, as it provides only limited search functionality. The default
+approach, where each subfield has its own entry in the mappings, works well in
+the majority of cases.
+
+A `json` field can be created as follows:
+[source,js]
+--------------------------------
+PUT bug_reports
+{
+  "mappings": {
+    "_doc": {
+      "properties": {
+        "title": {
+          "type": "text"
+        },
+        "labels": {
+          "type": "json"
+        }
+      }
+    }
+  }
+}
+
+POST bug_reports/_doc/1
+{
+  "title": "Results are not sorted correctly.",
+  "labels": {
+    "priority": "urgent",
+    "release": ["v1.2.5", "v1.3.0"],
+    "timestamp": {
+      "created": 1541458026,
+      "closed": 1541457010
+    }
+  }
+}
+--------------------------------
+// CONSOLE
+
```

Comment:
It might be better to mark this `// TESTSETUP` then you don't need to mark the others `// TEST[continued]`.

Reply:
👍